src/main.rs

```diff
@@ -229,9 +236,14 @@ fn main() {
             match c.unwrap() {
                 TermEvent::Key(Key::Char(c)) if chat_input_enabled => match c {
                     '\n' => {
-                        client.send_chat(chat_input.clone());
-                        chat_input = String::new();
-                        chat_input_enabled = false;
+                        if chat_input.is_empty() {
+                            chat_input = String::new();
```

Comment:
The second line here is unnecessary, and the third line can be moved out of the condition because it's executed in both cases.